### PR TITLE
feat: kill r processes

### PR DESCRIPTION
**Pull request by fdlk**

### CODE REVIEW

File: armadillo/src/main/java/org/molgenis/armadillo/ArmadilloSession.java [hunk 1]
```diff
@@ -61,20 +66,31 @@ void tryDetachRConnection(RConnection connection) {
...
-      return rSession.attach();
+      var connection = rSession.attach();
+      rSession = null;
+      return connection;
     } catch (RserveException err) {
       throw new ArmadilloSessionException("Could not attach connection to RSession", err);
     }
   }
 
-  public synchronized void sessionCleanup() {
```
Comment (fdlk) [inline armadillo/src/main/java/org/molgenis/armadillo/ArmadilloSession.java:72]: No longer needs to wait for evaluation to be finished, it kills the process instead!

**Comment by sidohaakma** [comment]
[Cleanup stale sessions on the Rserve environment](https://trello.com/c/Y3NuiRsk/214-cleanup-stale-sessions-on-the-rserve-environment)

File: r/src/main/java/org/molgenis/r/service/ProcessServiceImpl.java [hunk 2]
```diff
@@ -41,36 +43,57 @@
...
+    this.rExecutorService = rExecutorService;
   }
 
   @Override
   public int countRserveProcesses(RConnection connection) {
     try {
-      return ((REXP) connection.eval(COUNT_RSERVE_PROCESSES_COMMAND).asList().get(0)).asInteger();
-    } catch (RserveException | REXPMismatchException | ClassCastException e) {
+      return ((REXP)
+              rExecutorService.execute(COUNT_RSERVE_PROCESSES_COMMAND, connection).asList().get(0))
```
Comment (sidohaakma) [inline r/src/main/java/org/molgenis/r/service/ProcessServiceImpl.java:65]: Nice to use the RExecutorService here!

File: armadillo/src/main/java/org/molgenis/armadillo/ArmadilloSession.java [hunk 3]
```diff
@@ -61,20 +66,31 @@ void tryDetachRConnection(RConnection connection) {
   RConnection getRConnection() {
     try {
       if (rSession == null) {
-        return connectionFactory.createConnection();
+        var connection = connectionFactory.createConnection();
+        pid = processService.getPid(connection);
+        return connection;
```
Comment (sidohaakma) [inline armadillo/src/main/java/org/molgenis/armadillo/ArmadilloSession.java:71]: Maybe log the pid as well? We ports, but pid's would be useful as well concerning you can execute a api call to kill it?
Or do we use the connection to determine the pid using the actuator endpoints?

**Review by sidohaakma — CHANGES_REQUESTED** [review]

File: r/src/main/java/org/molgenis/r/service/ProcessServiceImpl.java (hunk 2 above)
Comment (fdlk) [reply r/src/main/java/org/molgenis/r/service/ProcessServiceImpl.java:65]: Yes, I'd forgotten all about it :D

**Review by sidohaakma — APPROVED** [review]

**Pull request merged**